Code/src/main/resources/catalog/catalog.xml

```diff
@@ -18,105 +35,584 @@
+        		Left-associative fold of structure.
+        	</function>
+        	<function name="foldr1" signature="(a -> a -> a) -> [a] -> a">
+        		A variant of foldr that has no base case, an thus may only be applied to non-empty structures.
+        	</function>
+        	<function name="foldl1" signature="(a -> a -> a) -> [a] -> a">
+        		A variant of foldl that has no base case, and thus may only be applied to non-empty structures.
+        	</function>
+        	<function name="null" signature="[a] -> Bool">
+        		Test whether the structure is empty. The default implementation is optimized for
+        		structures that are similar to cons-lists, because there is no general way to do better.
+        	</function>
+        	<function name="length" signature="[a] -> Int">
+        		Returns the size/length of a finite structure as an Int. The default implementation is
+        		optimized for structures that are similar to cons-lists, because there is no general way to
+        		do better.
+        	</function>
+        	<function name="elem" signature="Eq a => a - [a] -> Bool">
+        		Does the element occur in the structure?
+        	</function>
+        	<function name="maximum" signature="Ord a => [a] -> a">
+        		The largest element of a non-empty structure.
+        	</function>
+        	<function name="minimum" signature="Ord a => [a] -> a">
+        		The least element of a non-empty structure.
+        	</function>
+        	<function name="sum" signature="Num a => [a] -> a">
+        		The sum function computes the sum of the numbers of a structure.
+        	</function>
+        	<function name="product" signature="Num a => [a] -> a">
+        		The product function computes the product of the numbers of a structure.
+        	</function>
         </category>
-        <category name="Assorted">
-            <function name="id" signature="a -> a" >
-                Identity function.
-            </function>
-            <function name="const" signature="a -> b -> a">
-                Constant function.
-            </function>
-            <function name="undefined" signature="a">
-                Undefined function.
-            </function>
+        <category name="Miscellaneous functions">
+        	<function name="id" signature="a -> a">
+        		identity function.
+        	</function>
+        	<function name="const" signature="a -> b -> a">
+        		constant function.
+        	</function>
+        	<function name="(.)" signature="(b -> c) -> (a -> b) -> a -> c">
+        		Function composition
+        	</function>
+        	<function name="flip" signature="(a -> b -> c) -> b -> a -> c">
+        		flip f takes its (first) two arguments in the reverse order of f.
+        	</function>
+        	<function name="($)" signature="(a -> b) -> a -> b">
+        		Application operator. This operator is redundant, since ordinary application (f x) means
```

Comment:
Description doesn't make much sense in viskell: the operator is not redundant, it's the primary way to do function application.